### PR TITLE
Generate a Versions file to be consumed at runtime

### PR DESCRIPTION
**Pull request by andreaTP**
This will enable compile time checks of the versions in the Quarkus Operator SDK.
ref: https://github.com/quarkiverse/quarkus-operator-sdk/pull/460

cc. @metacosm

### CODE REVIEW

**Comment by metacosm** [comment]
Note that this should probably be integrated with https://github.com/java-operator-sdk/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/Version.java, which already provides the SDK version.

**Comment by andreaTP** [comment]
@metacosm I took a look and the scope of `Version` seems to be quite different, as well as the way to achieve it.
Do you have something in mind?

cc. @csviri

**Comment by metacosm** [comment]
> @metacosm I took a look and the scope of `Version` seems to be quite different, as well as the way to achieve it. Do you have something in mind?
> 
> cc. @csviri

What I meant is that all these versions should be accessible via `Version`… We could probably keep the current `Versions` however, it shouldn't be public, imo, and its information accessed via the `Version` API (which is also reused by the Quarkus extension).

**Comment by andreaTP** [comment]
@metacosm I have attempted to follow your comment, unsure if this is what you meant ...

File: operator-framework-core/src/main/java-templates/io/javaoperatorsdk/operator/api/config/Versions.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package io.javaoperatorsdk.operator.api.config;
+
+public final class Versions {
+
+  private Versions() {}
+
+  protected static final String JOSDK = "${project.version}";
```
Comment (csviri) [inline operator-framework-core/src/main/java-templates/io/javaoperatorsdk/operator/api/config/Versions.java:7]: Could you pls enlighten me how this supposed to work? Where are those placeholders filled?

File: operator-framework-core/pom.xml [hunk 2]
```diff
@@ -47,6 +47,19 @@
           <commitIdGenerationMode>full</commitIdGenerationMode>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
```
Comment (csviri) [inline operator-framework-core/pom.xml:51]: Why is this needed?

File: operator-framework-core/src/main/java-templates/io/javaoperatorsdk/operator/api/config/Versions.java (hunk 1 above)
Comment (andreaTP) [reply operator-framework-core/src/main/java-templates/io/javaoperatorsdk/operator/api/config/Versions.java:7]: This file is filled by the Maven Plugin as it is in `java-templates` folder as opposed to `java`.

File: operator-framework-core/pom.xml (hunk 2 above)
Comment (andreaTP) [reply operator-framework-core/pom.xml:51]: To fill in the information that is available in Maven.

**Review by csviri — APPROVED** [review]
I have no objections, if this helps with the quarkus extension fine by me.

File: operator-framework-core/src/main/java-templates/io/javaoperatorsdk/operator/api/config/Versions.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java-templates/io/javaoperatorsdk/operator/api/config/Versions.java:7]: would it be possible to write a unit test for this?

Comment (csviri) [reply operator-framework-core/src/main/java-templates/io/javaoperatorsdk/operator/api/config/Versions.java:7]: great, thx!

**Comment by metacosm** [comment]
@andreaTP could you give me access to push changes to your branch, please? Or would you prefer I open a different PR based on this one with my changes?

**Comment by andreaTP** [comment]
@metacosm as you prefer, you have an invite for my fork or you can open another PR to target the improvements or whatever.
No strong opinions at all! 👍

**Comment by metacosm** [comment]
Didn't easily manage to push changes to the original branch so created a new PR: #1706.

**Comment by andreaTP** [comment]
No worries, thanks @metacosm !
